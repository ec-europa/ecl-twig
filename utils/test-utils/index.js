const Twig = require('twig');
const { format } = require('prettier');

const renderTwigFile = (file, options, cb) =>
  Twig.renderFile(file, options, (err, html) =>
    cb(err, format(html, { parser: 'html' }))
  );

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) =>
    Twig.renderFile(file, options, (err, html) => {
      if (err) return reject(err);

      const div = document.createElement('div');
      div.innerHTML = html.trim();

      return resolve(div.firstChild);
    })
  );

module.exports = {
  renderTwigFile,
  renderTwigFileAsNode,
};
