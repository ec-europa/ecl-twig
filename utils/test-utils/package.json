{
  "name": "@ecl-twig/test-utils",
  "version": "2.29.0",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
