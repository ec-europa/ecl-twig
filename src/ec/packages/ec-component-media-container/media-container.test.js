import { merge, renderTwigFileAsNode } from '@ecl-twig/test-utils';

import demoContentImg from '@ecl/ec-specs-media-container/demo/data--image';
import demoContentVideo from './demo/data';

describe('EC Media Container', () => {
  const template =
    '@ecl-twig/ec-component-media-container/media-container.html.twig';
  const render = params => renderTwigFileAsNode(template, params);
  const defaultDataStructure = demoContentImg;

  describe('Media Container generic tests', () => {
    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const optionsWithExtraClasses = merge(defaultDataStructure, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(optionsWithExtraClasses)).resolves.toMatchSnapshot();
    });

    test('Media container renders correctly with extra attributes', () => {
      expect.assertions(1);

      const optionsWithExtraClasses = merge(defaultDataStructure, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(optionsWithExtraClasses)).resolves.toMatchSnapshot();
    });
  });

  describe('EC - Media container image', () => {
    test('Media container image renders correctly', () => {
      expect.assertions(1);
      return expect(render(defaultDataStructure)).resolves.toMatchSnapshot();
    });
  });

  describe('EC - Media container video', () => {
    test('Media container video renders correctly', () => {
      const options = merge(defaultDataStructure, {
        tracks: demoContentVideo.tracks,
        sources: demoContentVideo.sources,
      });

      expect.assertions(1);
      return expect(render(options)).resolves.toMatchSnapshot();
    });
  });
});
