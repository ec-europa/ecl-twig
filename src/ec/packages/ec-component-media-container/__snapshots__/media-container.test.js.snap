// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EC Media Container EC - Media container image Media container image renders correctly 1`] = `
<figure
  class="ecl-media-container"
>
  <img
    alt="Jean Monnet banner."
    class="ecl-media-container__media"
    src="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg"
  />
  <figcaption
    class="ecl-media-container__caption"
  >
    The European Commission has put forward ambitious yet realistic proposals for a modern EU budget. It is time for an EU budget that reflects rapid developments in innovation, the economy, the environment and geopolitics, amongst others.
  </figcaption>
</figure>
`;

exports[`EC Media Container EC - Media container video Media container video renders correctly 1`] = `
<figure
  class="ecl-media-container"
>
  <video
    class="ecl-media-container__media"
    controls=""
    poster="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg"
  >
    <source
      src="https://clips.vorwaerts-gmbh.de/big_buck_bunny.mp4"
      type="video/mp4"
    />
    <source
      src="https://clips.vorwaerts-gmbh.de/big_buck_bunny.webm"
      type="video/webm"
    />
    <track
      kind="captions"
      label="English"
      src="/captions/bunny-en.vtt"
      srclang="en"
    />
    <track
      kind="captions"
      label="franÃ§ais"
      src="/captions/bunny-fr.vtt"
      srclang="fr"
    />
  </video>
  <figcaption
    class="ecl-media-container__caption"
  >
    The European Commission has put forward ambitious yet realistic proposals for a modern EU budget. It is time for an EU budget that reflects rapid developments in innovation, the economy, the environment and geopolitics, amongst others.
  </figcaption>
</figure>
`;

exports[`EC Media Container Media Container generic tests Media container renders correctly with extra attributes 1`] = `
<figure
  class="ecl-media-container"
  data-test="data-test-value"
  data-test-1="data-test-value-1"
>
  <img
    alt="Jean Monnet banner."
    class="ecl-media-container__media"
    src="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg"
  />
  <figcaption
    class="ecl-media-container__caption"
  >
    The European Commission has put forward ambitious yet realistic proposals for a modern EU budget. It is time for an EU budget that reflects rapid developments in innovation, the economy, the environment and geopolitics, amongst others.
  </figcaption>
</figure>
`;

exports[`EC Media Container Media Container generic tests renders correctly with extra class names 1`] = `
<figure
  class="ecl-media-container custom-class custom-class--test"
>
  <img
    alt="Jean Monnet banner."
    class="ecl-media-container__media"
    src="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg"
  />
  <figcaption
    class="ecl-media-container__caption"
  >
    The European Commission has put forward ambitious yet realistic proposals for a modern EU budget. It is time for an EU budget that reflects rapid developments in innovation, the economy, the environment and geopolitics, amongst others.
  </figcaption>
</figure>
`;
